Tests/Opc.Ua.Core.Tests/Types/Encoders/JsonEncoderBenchmarks.cs

```diff
@@ -331,4 +335,791 @@ public void GlobalCleanup()
+        /// </summary>
+        private void EscapedStringLegacy(string value)
+        {
+            foreach (char ch in value)
+            {
+                bool found = false;
+
+                for (int ii = 0; ii < m_specialChars.Length; ii++)
+                {
+                    if (m_specialChars[ii] == ch)
+                    {
+                        m_streamWriter.Write('\\');
+                        m_streamWriter.Write(m_substitution[ii]);
+                        found = true;
+                        break;
+                    }
+                }
+
+                if (!found)
+                {
+                    if (ch < 32)
+                    {
+                        m_streamWriter.Write("\\u");
+                        m_streamWriter.Write("{0:X4}", (int)ch);
+                        continue;
+                    }
+
+                    m_streamWriter.Write(ch);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Version used previously in JsonEncoder plus improvement of binary encoding.
+        /// </summary>
+        /// <remarks>
+        /// For the underlying stream writer it is faster to write two chars than a 2 char string.
+        /// </remarks>
+        private void EscapedStringLegacyPlus(string value)
+        {
+            foreach (char ch in value)
+            {
+                bool found = false;
+
+                for (int ii = 0; ii < m_specialChars.Length; ii++)
+                {
+                    if (m_specialChars[ii] == ch)
+                    {
+                        m_streamWriter.Write('\\');
+                        m_streamWriter.Write(m_substitution[ii]);
+                        found = true;
+                        break;
+                    }
+                }
+
+                if (!found)
+                {
+                    if (ch < 32)
+                    {
+                        m_streamWriter.Write('\\');
+                        m_streamWriter.Write('u');
+                        m_streamWriter.Write(((int)ch).ToString("X4", CultureInfo.InvariantCulture));
+                        continue;
+                    }
+
+                    m_streamWriter.Write(ch);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Using a span to escape the string, write strings to stream writer if possible.
+        /// </summary>
+        private void EscapeStringSpan(string value)
+        {
+            ReadOnlySpan<char> charSpan = value.AsSpan();
+            int lastOffset = 0;
+
+            for (int i = 0; i < charSpan.Length; i++)
+            {
+                bool found = false;
+                char ch = charSpan[i];
+
+                for (int ii = 0; ii < m_specialChars.Length; ii++)
+                {
+                    if (m_specialChars[ii] == ch)
+                    {
+                        if (lastOffset < i)
+                        {
+#if NET48
```

Comment:
I *think* the right way is probably NETCOREAPP2_1_OR_GREATER and reverse

Reply:
updated